cpp/oneapi/dal/backend/primitives/reduction/test/reduction_dpc.cpp

```diff
@@ -0,0 +1,257 @@
+            ndview<float_t, 2, ndorder::c>::wrap(input_array.get_mutable_data(), { height, width });
+        auto output =
+            ndview<float_t, 1, ndorder::c>::wrap(output_array.get_mutable_data(), { height });
+
+        auto reduce_event =
+            reduce_by_rows(get_queue(), input, output, binary_t{}, unary_t{}, { out_event });
+        reduce_event.wait_and_throw();
+
+        check_output_rm_rw(output_array);
+    }
+
+    void test_rm_cw_reduce() {
+        auto input_array = row_accessor<const float_t>{ input_table }.pull(get_queue());
+        auto [output_array, out_event] = output(width);
+        auto input =
+            ndview<float_t, 2, ndorder::c>::wrap(input_array.get_mutable_data(), { height, width });
+        auto output =
+            ndview<float_t, 1, ndorder::c>::wrap(output_array.get_mutable_data(), { width });
+
+        auto reduce_event =
+            reduce_by_columns(get_queue(), input, output, binary_t{}, unary_t{}, { out_event });
+        reduce_event.wait_and_throw();
+
+        check_output_rm_cw(output_array);
+    }
+
+    void test_cm_cw_reduce() {
+        auto input_array = row_accessor<const float_t>{ input_table }.pull(get_queue());
+        auto [output_array, out_event] = output(height);
+        auto input_tr =
+            ndview<float_t, 2, ndorder::c>::wrap(input_array.get_mutable_data(), { height, width });
+        auto input = input_tr.t();
+        auto output =
+            ndview<float_t, 1, ndorder::c>::wrap(output_array.get_mutable_data(), { height });
+
+        auto reduce_event =
+            reduce_by_columns(get_queue(), input, output, binary_t{}, unary_t{}, { out_event });
+        reduce_event.wait_and_throw();
+
+        check_output_rm_rw(output_array);
+    }
+
+    void test_cm_rw_reduce() {
+        auto input_array = row_accessor<const float_t>{ input_table }.pull(get_queue());
+        auto [output_array, out_event] = output(width);
+        auto input_tr =
+            ndview<float_t, 2, ndorder::c>::wrap(input_array.get_mutable_data(), { height, width });
+        auto input = input_tr.t();
+        auto output =
+            ndview<float_t, 1, ndorder::c>::wrap(output_array.get_mutable_data(), { width });
+
+        auto reduce_event =
+            reduce_by_rows(get_queue(), input, output, binary_t{}, unary_t{}, { out_event });
+        reduce_event.wait_and_throw();
+
+        check_output_cm_rw(output_array);
+    }
+
+    auto get_width() const {
+        return width;
+    }
+
+    auto get_height() const {
+        return height;
+    }
+
+private:
+    const binary_t binary{};
+    const unary_t unary{};
+
+private:
+    std::int64_t width;
+    std::int64_t height;
+    table input_table;
+};
+
+TEMPLATE_LIST_TEST_M(reduction_test_random,
+                     "Randomly filled Row-Major Row-Wise reduction",
+                     "[reduction][rm][small]",
+                     reduction_types) {
+    this->generate();
+    this->test_rm_rw_reduce();
```

Comment:
I've tried to launch it by myself, on architecture that does not support `double` natively, reduction tests take > 2 minutes, it's more than all other tests together. It's not acceptable:
1. Reduce parameters enumeration
2. Add `SKIP_IF` if there is no native support for `double`, [see here](https://github.com/oneapi-src/oneDAL/blob/4e0570d6ce8e43843e3b5547408c8963689c941b/cpp/oneapi/dal/backend/primitives/stat/test/cov_dpc.cpp#L96)